blocks/graph.py

```diff
@@ -37,6 +37,7 @@ def _get_variables(self):
         self.applies = set()
         self.application_calls = set()
         self.updates = []
+        self.auxiliary_variables = set()
```

Comment:
Wait, why wouldn't it attach it as an auxiliary variable? My idea was this: All the linear bricks automatically assign their weight decays (L1, L2 norms) as auxiliary variables. When you want to apply weight decay you would simply query the computation graph (or use selectors?) to give you the relevant weight decays which you can then add to the cost you are training on. What I haven't really figured out is how we would filter out a particular weight decay if we only want to use it for 1 brick.